### PR TITLE
show message about server error when sync is active

### PR DESCRIPTION
**Pull request by cezaraugusto**
fix https://github.com/brave/brave-browser/issues/2577

Test Plan:

1. Create a sync chain and ensure you see two devices
2. Quit Brave
3. Launch Brave without internet
4. You should see a Unable to connect to the Sync servers. error dialog

This also adds the internet connection error

### CODE REVIEW

File: components/brave_sync/ui/components/enabledContent.tsx [hunk 1]
```diff
@@ -111,6 +111,10 @@ export default class SyncEnabledContent extends React.PureComponent<Props, State
     })
   }
 
+  onUserNoticedError = () => {
+    this.props.actions.resetSyncSetupError()
```
Comment (darkdh) [inline components/brave_sync/ui/components/enabledContent.tsx]: SYNC_RESET_SETUP_ERROR, this naming is confusing. what exactly is it related to reset?

Comment (cezaraugusto) [reply components/brave_sync/ui/components/enabledContent.tsx]: this dispatches a message cleaning up the error message (store receives `undefined`)

Comment (darkdh) [reply components/brave_sync/ui/components/enabledContent.tsx]: per discussion in DM, please do a follow-up for better naming of this every where.

**Comment by cezaraugusto** [comment]
master c4b0aec
0.60.x 11b82711
0.59.x uplift request

**Comment by rebron** [comment]
Approving uplift-request for 0.59.x-beta. cc: @kjozwiak cc: @srirambv

**Pull request merged**